hpy/devel/src/runtime/ctx_tracker.c

```diff
@@ -0,0 +1,154 @@
+/**
+ * A manager for HPy handles, allowing handles to be tracked
+ * and closed as a group.
+ *
+ * Note::
+ *    Internally, HPyTracker always keeps space for one extra handle so that
+ *    HPyTracker_Add can always store the handle being passed to it,
+ *    even if it fails to automatically create space for future
+ *    handles. This allows HPyTracker_Free to close all handles passed to
+ *    HPyTracker_Add.
+ *
+ *    Space for this extra handle is created automatically, so
+ *    HPyTracker_NewWithSize(ctx, 0) will actually allocated space for one
+ *    handle.
+ *
+ *    Calling HPyTracker_NewWithSize(ctx, n) will ensure that at least n handles
+ *    can be tracked without the need for a resize.
+ *
+ * Example usage (inside an HPyDef_METH function)::
+ *
+ * long i;
+ * HPy key, value;
+ * HPyTracker hl;
```

Comment:
Minor issue, but this still has an old `HList` variable name. Maybe `trk` or `track`?

Reply:
Good catch! I renamed them to `ht` for `HPyTracker`.